quadratic/base.py

```diff
@@ -61,13 +61,21 @@ def min_ineq_qp(problem: QuadraticProblem) -> np.ndarray:
         if np.all(p == 0):
             A, _ = combine_linear([eq.c for eq in working_set])
             lambda_vec = np.linalg.solve(A, g)
-            if np.all(lambda_vec >= 0):
+
+            # in the working set we transformed all constraints into equalities, but we want to check for inequalities
+            current_set = [constr for constr in problem.constraints if
+                           np.any([constr.equal_callables(working_constr) for working_constr in working_set])]
+            _is_ineq_constr = [eq.equation_type != EquationType.EQ for eq in current_set]
+            lambda_vec = lambda_vec[_is_ineq_constr]
+
+            if np.all(lambda_vec >= 0) and len(lambda_vec) != 0:
                 return x
             else:
                 least_lambda_index = np.argmin(lambda_vec)
                 del working_set[least_lambda_index]
         else:
-            blocking_constraints = [constraint for constraint in active_set if constraint not in working_set]
+            # blocking constraints are all constraints of the problem that are not in the working set
+            blocking_constraints = [constraint for constraint in problem.constraints if constraint not in working_set]
```

Comment:
Das wird vermutlich nimmer funktionieren, weil ma de __eq__ jetz removed haben und jetz de default __eq__ implementierung anschlägt - und da is dann der Unterschied mit dem equality_type schlagend.
Müsste ähnlich wie oben gelöst werden